src/BaseObject.php

```diff
@@ -6,46 +6,64 @@
 
 require_once __DIR__ . '/../boot.php';
 
+use Dice\Dice;
 use Friendica\Network\HTTPException\InternalServerErrorException;
 
 /**
  * Basic object
  *
  * Contains what is useful to any object
+ *
+ * Act's like a global registry for classes
  */
 class BaseObject
 {
 	/**
-	 * @var App
+	 * @var Dice The Dependency Injection library
 	 */
-	private static $app = null;
+	private static $dice;
+
+	/**
+	 * Set's the dependency injection library for a global usage
+	 *
+	 * @param Dice $dice The dependency injection library
+	 */
+	public static function setDependencyInjection(Dice $dice)
+	{
+		self::$dice = $dice;
+	}
 
 	/**
 	 * Get the app
 	 *
 	 * Same as get_app from boot.php
 	 *
 	 * @return App
-	 * @throws \Exception
 	 */
 	public static function getApp()
 	{
-		if (empty(self::$app)) {
-			throw new InternalServerErrorException('App isn\'t initialized.');
-		}
-
-		return self::$app;
+		return self::getClass(App::class);
 	}
 
 	/**
-	 * Set the app
+	 * Returns the initialized class based on it's name
+	 *
+	 * @param string $name The name of the class
 	 *
-	 * @param App $app App
+	 * @return object The initialized name
 	 *
-	 * @return void
+	 * @throws InternalServerErrorException
 	 */
-	public static function setApp(App $app)
+	protected static function getClass(string $name)
 	{
-		self::$app = $app;
+		if (empty(self::$dice)) {
+			throw new InternalServerErrorException('DICE isn\'t initialized.');
+		}
+
+		if (class_exists($name) || interface_exists($name   )) {
+			return self::$dice->create($name);
```

Comment:
This won't create different App object for each call to `getApp()`, right?

Reply:
Nope, because of #7427 , every class will be reused. Before #7427 , I forgot about it and created a massive performance-problem with some strange side-effects during dev, so this global setting is indeed important ;-)